x/stakeibc/keeper/unbonding_records.go

```diff
@@ -278,133 +271,124 @@ func (k Keeper) CleanupEpochUnbondingRecords(ctx sdk.Context) bool {
+			if err != nil {
+				k.Logger(ctx).Error(err.Error())
+				return false, 0
+			}
 
-				// Send the transaction through SubmitTx
-				_, err = k.SubmitTxsDayEpoch(ctx, hostZone.ConnectionId, msgs, *delegationAccount, REDEMPTION, marshalledCallbackArgs)
-				if err != nil {
-					ctx.Logger().Info(fmt.Sprintf("Failed to SubmitTxs, transfer to redemption account on %s", hostZone.ChainId))
-				}
-				ctx.Logger().Info(fmt.Sprintf("Successfully completed unbonded token sweep ICA call for %s, %s, %v", hostZone.ConnectionId, hostZone.ChainId, msgs))
-			} else {
-				k.Logger(ctx).Info(fmt.Sprintf("\tNot sweeping tokens for host zone %s because redemption/delegation accounts aren't registered", hostZone.ChainId))
-				success = false
-				failed_sweeps = append(failed_sweeps, hostZone.ChainId)
-				return nil
+			// Send the transaction through SubmitTx
+			_, err = k.SubmitTxsDayEpoch(ctx, hostZone.ConnectionId, msgs, *delegationAccount, REDEMPTION, marshalledCallbackArgs)
+			if err != nil {
+				ctx.Logger().Info(fmt.Sprintf("Failed to SubmitTxs, transfer to redemption account on %s", hostZone.ChainId))
 			}
+			ctx.Logger().Info(fmt.Sprintf("Successfully completed unbonded token sweep ICA call for %s, %s, %v", hostZone.ConnectionId, hostZone.ChainId, msgs))
 		} else {
-			k.Logger(ctx).Info(fmt.Sprintf("\tNo unbonded tokens this day to sweep for host zone %s", hostZone.ChainId))
+			k.Logger(ctx).Info(fmt.Sprintf("\tNot sweeping tokens for host zone %s because redemption/delegation accounts aren't registered", hostZone.ChainId))
+			return false, 0
+		}
+	} else {
+		k.Logger(ctx).Info(fmt.Sprintf("\tNo unbonded tokens this day to sweep for host zone %s", hostZone.ChainId))
+	}
+	return true, totalAmtTransferToRedemptionAcct
+}
+
+func (k Keeper) SweepAllUnbondedTokens(ctx sdk.Context) (success bool, successfulSweeps []string, sweepAmounts []int64, failedSweeps []string) {
+	// this function returns true if all chains succeeded, false otherwise
+	// it also returns a list of successful chains (arg 2), tokens swept (arg 3), and failed chains (arg 4)
+	success = true
+	successfulSweeps = []string{}
+	sweepAmounts = []int64{}
+	failedSweeps = []string{}
+	hostZones := k.GetAllHostZone(ctx)
+	for _, hostZone := range hostZones {
+		hostZoneSuccess, sweepAmount := k.SweepAllUnbondedTokensForHostZone(ctx, hostZone)
+		if hostZoneSuccess {
+			successfulSweeps = append(successfulSweeps, hostZone.ChainId)
+			sweepAmounts = append(sweepAmounts, sweepAmount)
+		} else {
+			success = false
+			failedSweeps = append(failedSweeps, hostZone.ChainId)
 		}
-		successful_sweeps = append(successful_sweeps, hostZone.ChainId)
-		sweep_amounts = append(sweep_amounts, totalAmtTransferToRedemptionAcct)
-		return nil
 	}
-	// Iterate the zones and sweep their unbonded tokens
-	k.IterateHostZones(ctx, sweepUnbondedTokens)
-	return success, successful_sweeps, sweep_amounts, failed_sweeps
+	return success, successfulSweeps, sweepAmounts, failedSweeps
```

Comment:
are multiple values returned just for the unit tests? Or are these values leveraged in hooks

Reply:
just for the unit tests, these are ignored in hooks right now